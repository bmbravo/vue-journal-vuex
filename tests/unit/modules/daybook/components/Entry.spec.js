import { shallowMount } from '@vue/test-utils';
import EntryComponent from '@/modules/daybook/components/EntryComponent';
import { journalState } from '../../../mock-data/test-journal-state';

describe('Pruebas en el entry component', () => {
  const mockRouter = {
    push: jest.fn(),
  };

  const wrapper = shallowMount(EntryComponent, {
    props: {
      entry: journalState.entries[1],
    },
    global: {
      mocks: {
        $router: mockRouter,
      },
    },
  });

  test('Debe de hacer match con el snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('Debe de redireccionar al hacer click en el entry-container', () => {
    const entryContainer = wrapper.find('.entry-container');

    entryContainer.trigger('click');

    expect(mockRouter.push).toHaveBeenCalled();
    expect(mockRouter.push).toHaveBeenCalledWith({
      name: 'entry',
      params: { id: journalState.entries[1].id },
    });
  });

  test('Pruebas en las propiedades computadas', () => {
    expect(wrapper.vm.day).toBe(10);
    expect(wrapper.vm.month).toBe('Julio');
    expect(wrapper.vm.yearDay).toBe('2023, Lunes');
  });
});
