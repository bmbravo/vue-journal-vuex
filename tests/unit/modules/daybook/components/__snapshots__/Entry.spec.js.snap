// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el entry component Debe de hacer match con el snapshot 1`] = `
<div class="entry-container mb-3 pointer p-2">
  <!-- title -->
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">10</span><span class="mx-1 fs-5">Julio</span><span class="mx-2 fw-light">2023, Lunes</span></div>
  <div class="entry-description">Esto es una prueba

    Dolore est quis cillum aute do adipisicing laborum. Ut esse proident excepteur reprehenderit velit anim laboru...</div>
</div>
`;
